### ISSUE TITLE
[Bug (Crash)] Drawer crash on search

### ISSUE THREAD

**Issue by Reprevise** [Deletescape-Media/Lawnchair#1810]
## Description

When I go to search for an app, the launcher crashes and it reverts back to my Samsung Launcher.
 
**NOTE:** On the beta version of Lawnchair on the Play Store, I do not get the same error/bug. (Version: 2.0-2551 as of 11/16/2019). I do not know what is causing this issue. I thought at first it was an icon pack but it definitely could be a hidden apps issue as @matty1086 suggested.

## Expected Behavior

When I search for an app, it should filter out the apps and bring me a list that contains the characters which I searched for.

## Actual Behavior

Instead, when I search for an app, I don't get the app and it says Lawnchair crashed and gives me a log.

## Steps to Reproduce

https://youtu.be/-cuJc7vzvew

## Environment

* Device: Samsung Galaxy S10
* Android version: 9.0
* Launcher version: 2.0-2531-ci-q-prealpha
* Rom: One UI 1.1

## Logcat

 

```
Lawnchair bug report Nov 12, 2019 8:50:32 AM
Lawnchair version: 2.0-2531-ci-q-pre-alpha (2531)
build.brand: samsung
build.device: beyond1q
build.display: PPR1.180610.011.G973USQU2BSIP
build.fingerprint: samsung/beyond1qltesq/beyond1q:9/PPR1.180610.011/G973USQU2BSIP:user/release-keys
build.hardware: qcom
build.id: PPR1.180610.011
build.manufacturer: samsung
build.model: SM-G973U
build.product: beyond1qltesq
build.type: user
version.codename: REL
version.incremental: G973USQU2BSIP
version.release: 9
version.sdk_int: 28

error: Uncaught exception

--------- beginning of stacktrace
java.lang.IllegalStateException: Cache accessed on wrong thread Looper (main, tid 2) {56e9a38}
	at com.android.launcher3.icons.cache.BaseIconCache.assertWorkerThread(BaseIconCache.java:560)
	at com.android.launcher3.icons.cache.BaseIconCache.cacheLocked(BaseIconCache.java:316)
	at com.android.launcher3.icons.cache.BaseIconCache.cacheLocked(BaseIconCache.java:308)
	at com.android.launcher3.icons.IconCache.getTitleAndIcon(IconCache.java:210)
	at com.android.launcher3.icons.IconCache.getTitleAndIcon(IconCache.java:191)
	at com.android.launcher3.allapps.AlphabeticalAppsList.getFiltersAppInfos(AlphabeticalAppsList.java:555)
	at com.android.launcher3.allapps.AlphabeticalAppsList.refillAdapterItems(AlphabeticalAppsList.java:433)
	at com.android.launcher3.allapps.AlphabeticalAppsList.updateAdapterItems(AlphabeticalAppsList.java:376)
	at com.android.launcher3.allapps.AlphabeticalAppsList.onAppsUpdated(AlphabeticalAppsList.java:368)
	at com.android.launcher3.allapps.AlphabeticalAppsList.setOrderedFilter(AlphabeticalAppsList.java:295)
	at com.google.android.apps.nexuslauncher.qsb.FallbackAppsSearchView.onSearchResult(FallbackAppsSearchView.java:54)
	at com.google.android.apps.nexuslauncher.search.SearchThread.handleMessage(SearchThread.java:97)
	at android.os.Handler.dispatchMessage(Handler.java:102)
	at android.os.Looper.loop(Looper.java:214)
	at android.app.ActivityThread.main(ActivityThread.java:7037)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:494)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:965)
```

**Comment by Reprevise** [comment]
Huh, seems to only happen with an icon pack enabled.

**Update**: I disabled the icon pack in Lawnchair and downloaded another one from the play store. I enabled it and still the same issue occured. Now this time, I couldn't disable it. I would drag it down and back out but it still showed as enabled. I have uninstalled the icon pack and it says its using system icons. However I my home screen is still using the icons from the pack I downloaded. I pressed "Reset custom icons" but that didn't do anything.

**Update 2**: I had another icon pack that was disabled in Lawnchair so I enabled it and went to my home screen to check if the icons changed. Then I went back and disabled it them cleared the custom icons which seemed to work. Now the drawer search seems to work as normal. The icon pack that I was using that wouldn't disable is called "Minma Icon Pack" (the free one).

**Comment by Reprevise** [comment]
Okay, maybe I was wrong. I have no icon packs enabled and Lawnchair just crashed with the same error. I don't know what's causing it anymore. I thought it was the icon packs but I guess not.

**Comment by deafeningquiet** [comment]
Ive been getting the same crash since the last update and think I figured out that it occurs when a hidden app should show in the results.

I'm able to search for most apps, however, once I search the name of the app I have hidden from drawer, the crash occurs.

**Comment by Reprevise** [comment]
What version of Lawnchair are you running @matty1086? I have recently just switched to the Play Store beta version of Lawnchair (2.0-2551) and it seems to work fine now. I assume you're on one of the Android Q (10) builds?

**UPDATE:** I have downloaded the most recent build of Lawnchair (2546) and I set up a hidden app on my device. When I allow the search of that app in the drawer and then search for the app in the drawer, Lawnchair crashes. @matty1086 is correct, hidden apps apparently cause the most recent alpha build of Lawnchair to crash.
